BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/distance/BeaconDistanceCalculator.java

```diff
@@ -21,6 +21,8 @@
 
     public static final int SIGNAL_LOSS_AT_ONE_METER = -41;
 
+    private static float pathLossParameter = PATH_LOSS_PARAMETER_OFFICE_HARD_PARTITION;
```

Comment:
Remove `static` keyword (also from the getter & setter).

Reply:
I don't see any issue with having non-static fields in an abstract class. Check out the `BeaconLocationProvider` for a reference.

Reply:
You cannot use a non-static fields inside a static methods, you will need to make it static, because you can get instances from this class, what field value you will return if you have multiple instances?

You are using the BeaconLocationProvider class by making a child class from it, like what you do in the debugging provider in the sample, in this way you can set fields as protected or public and access them inside the child class.

I guess you have maid this class as abstract because you don't want anyone to make an instance from it and you want to use calculateDistance method statically.

Reply:
You're right, I missed the context here.